src/compiler/crystal/tools/init.cr

```diff
@@ -168,10 +174,15 @@ module Crystal
         @skip_existing = false
       )
       end
+
+      getter expanded_dir : ::Path { ::Path.new(dir).expand(home: true) }
+
+      getter github_repo : String { "#{github_name}/#{expanded_dir.basename}" }
```

Comment:
Since the computation here depends on the other properties I'd keep those as a regular methods with no memoization.

Reply:
Technically you're completely right. But `Config` properties are populated once and then never changed. This is not expressed in an explicit contract, but in the limited scope of its usage, I believe this is fine.
IMO it doesn't really matter one way or the other.